Core/FileSystems/BlockDevices.cpp

```diff
@@ -520,3 +532,149 @@ bool NPDRMDemoBlockDevice::ReadBlock(int blockNumber, u8 *outPtr, bool uncached)
 
 	return true;
 }
+
+/*
+ * CHD file
+ */
+static const UINT8 nullsha1[CHD_SHA1_BYTES] = { 0 };
+
+struct CHDImpl {
+	chd_file *chd = nullptr;
+	const chd_header *header = nullptr;
+};
+
+CHDFileBlockDevice::CHDFileBlockDevice(FileLoader *fileLoader)
+	: BlockDevice(fileLoader), impl_(new CHDImpl())
+{
+	// Default, in case of failure
+	numBlocks = 0;
+
+	chd_header childHeader;
+
+	int depth = 0;
+	Path paths[8];
+	paths[0] = fileLoader->GetPath();
+
+	chd_error err = chd_read_header(paths[0].c_str(), &childHeader);
```

Comment:
Doesn't this mean it won't work on Android?  Is this function even needed?  Later it opens it from a FILE * anyway, which would probably also solve the international path thing too?

Of course it won't support read ISO into RAM or remote disc streaming, but I guess people using this format mostly want the hype and will probably feel like it's more powerful because stuff doesn't work.

-[Unknown]

Reply:
That is absolutely true, but as commented elsewhere, I haven't enabled this path yet.

Reply:
This is at the very beginning of the function.  It's not a commented out part.

-[Unknown]

Reply:
Oh duh, was thinking of the below block, too lazy to click into the full view..

Yeah, this is bad. I just realize I only tested it on an old Android.

So this will need fixing for modern Android, and I'll probably end up submitting a patch to libchdr to add chd_read_header_file.

Reply:
Does it even need this though?  It seems to just open from a file and actually get the header from there later?

-[Unknown]

Reply:
True that, I'll look into getting rid of it before merge.